model/sharing/oauth.go

```diff
@@ -435,6 +435,16 @@ func (s *Sharing) ProcessAnswer(inst *instance.Instance, creds *APICredentials)
 			}
 			ac.Credentials.AccessToken = token
 
+			// Update the contact to fill the name if missing
+			if email := s.Members[i+1].Email; email != "" {
+				if c, err := contact.FindByEmail(inst, email); err == nil {
```

Comment:
This might be unrelated but is `FindByEmail()` resilient to case differences? Since we could have uppercase letters in user input, we could miss the contact.

Reply:
`FindByEmail()` is case sensitive, but we don't care here, because we will use the exact same string: the member email has been taken from the contact, and we use it to find the contact later.